src/test/java/de/numcodex/sq2cql/model/structured_query/TimeRestrictionTest.java

```diff
@@ -0,0 +1,51 @@
+package de.numcodex.sq2cql.model.structured_query;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import static java.time.format.DateTimeFormatter.ISO_DATE;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class TimeRestrictionTest {
+
+    @Test
+    void fromJsonNode() {
```

Comment:
Can you please write one test for each scenario? Also please use AssertJ assertions. I like to move towards them. The text from assertDoesNotThrow can go into @DisplayName of the individual tests. 

On top of that you do not test `fromJsonNode` at all. Instead, you test `create`. You can use Run with Code Coverage in IntelliJ to see that.